aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java

```diff
@@ -139,6 +149,105 @@ SdkIterable<GroupNameAndArn> getThingGroupIterable(String coreThingName) {
+     * compoenent.
+     *
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
+    @When("I install the component {word} from local store with configuration")
+    public void installComponentWithConfiguration(final String componentName, final String configurationTable)
+            throws InterruptedException, IOException {
+        // TODO: recipe of json format will also be taken.
+        // read the recipe from local store, and get component name and version from recipe
+        List<String> componentSpecs = Arrays.asList(
+                componentName, LOCAL_STORE_RECIPES.resolve(String.format("%s.yaml", componentName)).toString()
+        );
+
+        // handle the config input
+        Map<String, Object> configuration = readConfiguration(configurationTable);
+        CommandInput command = getCliDeploymentCommand(new ArrayList<>(Collections.singleton(componentSpecs)),
+                configuration);
+        createLocalDeploymentWithConfigs(0, command);
+    }
+
+    private Map<String, Object> readConfiguration(String configurationTable) throws JsonProcessingException {
+        String updatedConfiguration = this.scenarioContext.applyInline(configurationTable);
+        return this.mapper.readValue(updatedConfiguration,
+                new TypeReference<Map<String, Object>>() {});
+    }
+
+    private CommandInput getCliDeploymentCommand(List<List<String>> componentNames,
+                                         Map<String, Object> configuration) throws InterruptedException, IOException {
+        // find the component artifacts and copy into a local store
+        final Map<String, ComponentDeploymentSpecification> components = parseComponentNamesAndPrepare(componentNames);
+
+        List<String> commandArgs = new ArrayList<>();
+
+        commandArgs.addAll(Arrays.asList("deployment", "create",
+                "--artifactDir "  + artifactPath.toString(),
+                "--recipeDir " + recipePath.toString()));
+
+        for (Map.Entry<String, ComponentDeploymentSpecification> entry : components.entrySet()) {
+            String componentName = entry.getKey();
+            commandArgs.add(String.format("--merge %s=%s", componentName, entry.getValue().componentVersion()));
+            String updateConfigArgs = getCliUpdateConfigArgs(componentName, configuration);
+            if (!updateConfigArgs.isEmpty()) {
+                commandArgs.add("--update-config '" + updateConfigArgs + "'");
+            }
+        }
+        return CommandInput.builder()
+                .line(testContext.installRoot().resolve("bin").resolve("greengrass-cli").toString())
```

Comment:
nit: Worth extracting the duplicated code to a separate method. https://github.com/aws-greengrass/aws-greengrass-testing/blob/1ce7ef461465fb11f701f2e1231caa817afcba04/aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java#L171